### PR TITLE
[UX] Add footer throughout #949

### PR DESCRIPTION
**Pull request by Priyanshu44**
# Description

Adding custom footer throughout with static content.
 
image
 
Please note: For now, we removed empty links as discussed.

### CODE REVIEW

File: apps/client/src/layout/components/footer/footer.css [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/**Custom footer component CSS*/
+
+.dashboard-footer-container { 
+    z-index: 1002; 
+    position:sticky;
+    bottom:0;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    padding: 8px 30px;
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/footer.css:10]: use design token and inline styles for these pixels

File: apps/client/src/layout/components/footer/footer.css [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    display: flex;
+    flex-wrap: wrap;
+    list-style: none;
+    margin: 0;
+    padding: 0;
+    gap: 24px;
+}
+
+.dashboard-footer-container ul li a {
+    color: #ffff !important;
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/footer.css:24]: use design tokens

File: apps/client/src/layout/components/footer/footer.css [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+.dashboard-footer-container ul li a:hover {
+    color: #9ba7b6 !important;
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/footer.css:28]: same here

File: apps/client/src/layout/components/footer/footer.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { FormattedMessage } from 'react-intl';
+import './footer.css'
+
+export function Footer() {
+    const footerStyle = {
+        color: awsui.colorBackgroundControlDefault,
+        backgroundColor: awsui.colorBackgroundLayoutToggleDefault
+      };
+  return (
+    <footer id='f' className="dashboard-footer-container" style={footerStyle} data-testid="footer-component">
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/footer.tsx:12]: dont use f.. instead use something like 'app-footer'

**Review by mnischay — CHANGES_REQUESTED** [review]

File: apps/client/src/layout/components/footer/footer.css (hunk 3 above)
Comment (Priyanshu44) [reply apps/client/src/layout/components/footer/footer.css:28]: to style hover with inline CSS in React we need to add additional wrapper component and conditionally set inline styles using state on onMouseEnter and onMouseLeave events. Do we really need that ? @mnischay

File: apps/client/src/layout/components/footer/footer.css (hunk 1 above)
Comment (Priyanshu44) [reply apps/client/src/layout/components/footer/footer.css]: Resolved

File: apps/client/src/layout/components/footer/constants.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import * as awsui from '@cloudscape-design/design-tokens';
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/constants.ts:1]: `import { colorBackgroundControlDefault, .... } from ....`

File: apps/client/src/layout/components/footer/constants.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+interface IColor {color: string}
+
+interface IFooterContent {cloudshell : IColor, feedback: IColor, language: IColor, privacy: IColor, terms: IColor, cookie: IColor}
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/constants.ts:5]: why does all these need to be object ?

File: apps/client/src/layout/components/footer/footer.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
+import { useState } from 'react';
+import Link from "@cloudscape-design/components/link";
+import * as awsui from '@cloudscape-design/design-tokens';
+import { FormattedMessage } from 'react-intl';
+import { footerColor, Links } from './constants'
+import './footer.css'
+
+export function Footer() {
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/footer.tsx:8]: use arrow function

File: apps/client/src/layout/components/footer/footer.tsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
+import { useState } from 'react';
+import Link from "@cloudscape-design/components/link";
+import * as awsui from '@cloudscape-design/design-tokens';
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/footer.tsx:3]: please use name imports

**Review by mnischay — CHANGES_REQUESTED** [review]

File: apps/client/src/layout/components/footer/constants.ts (hunk 5 above)
Comment (Priyanshu44) [reply apps/client/src/layout/components/footer/constants.ts]: resolved

File: apps/client/src/layout/components/footer/constants.ts (hunk 6 above)
Comment (Priyanshu44) [reply apps/client/src/layout/components/footer/constants.ts]: resolved

File: apps/client/src/layout/components/footer/footer.tsx (hunk 7 above)
Comment (Priyanshu44) [reply apps/client/src/layout/components/footer/footer.tsx]: resolved

File: apps/client/src/layout/components/footer/footer.tsx (hunk 8 above)
Comment (Priyanshu44) [reply apps/client/src/layout/components/footer/footer.tsx]: resolved

File: apps/client/src/layout/components/footer/footer.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+    const gapStyle = {
+        gap: spaceScaledXl
+    }
+    
+  return (
+    <footer id='app-footer' className="dashboard-footer-container" style={footerStyle} data-testid="footer-component">
+            <ul style={gapStyle}>
+                <li>
+                    <Link  data-testid="cloudshell-link" href="#">
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/footer.tsx:34]: Sorry, i missed this before, can we have an array like 

`const someArray = [{testId: 'cloudshell-link', link: 'cloudshell', description: 'Open Cloudshell link'}, { feedback.... }]
`
and map thro in the html


```
<ul>
 { someArray.map(data =>  ( <li> ..... {data.description } </li>))}
</ui>
```

Comment (Priyanshu44) [reply apps/client/src/layout/components/footer/footer.tsx]: resolved

File: apps/client/src/layout/components/footer/constants.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+export const footerColor : { [key in string]: {color?: string} }  = {
+    cloudshell:{color:colorBackgroundControlDefault},
+    feedback:{color:colorBackgroundControlDefault},
+    language:{color:colorBackgroundControlDefault},
+    privacy: {color: colorBackgroundControlDefault},
+    terms: {color: colorBackgroundControlDefault},
+    cookie: {color: colorBackgroundControlDefault}
+}
+
+export const footerLeftContent: [{[key in string]: string},{[key in string]: string}, {[key in string]: string}] = [
```
Comment (mnischay) [inline apps/client/src/layout/components/footer/constants.ts:12]: declare a type
`const type footerContent = { name: string, test-id ..... } 
`
`const footerRightContent: footerContent[] = [ ] `

Comment (Priyanshu44) [reply apps/client/src/layout/components/footer/constants.ts]: resoled

**Review by mnischay — APPROVED** [review]

**Pull request merged**